Assets/Game/Addons/WorldDataEditor/Editor/WorldDataEditor.cs

```diff
@@ -670,24 +790,17 @@ private void GUIElementFlatSwitchType(ref WorldDataEditorObject data, ref int el
                     data.id = tempArray[pos];
                     int archive = int.Parse(data.id.Split('.')[0]);
                     int record = int.Parse(data.id.Split('.')[1]);
-                    if (dataMode == DataMode.Building)
-                    {
-                        DFBlock.RmbBlockFlatObjectRecord newBlockRecord = new DFBlock.RmbBlockFlatObjectRecord();
-                        newBlockRecord.TextureArchive = archive;
-                        newBlockRecord.TextureRecord = record;
-                        GameObject go = CreateFlatObject(newBlockRecord, isExteriorMode);
-                        data.gameObject.GetComponent<MeshRenderer>().sharedMaterials = go.GetComponent<MeshRenderer>().sharedMaterials;
-                        data.gameObject.GetComponent<MeshFilter>().sharedMesh = go.GetComponent<MeshFilter>().sharedMesh;
-                        data.gameObject.name = go.gameObject.name;
-                        DestroyImmediate(go);
-                    }
-                    else
-                    {
-                        data.rdbObj.Resources.FlatResource.TextureArchive = archive;
-                        data.rdbObj.Resources.FlatResource.TextureRecord = record;
-                        Billboard billboard = Selection.activeGameObject.GetComponent<Billboard>();
-                        billboard.SetMaterial(archive, record);
-                    }
+                    
+                    Billboard billboard = Selection.activeGameObject.GetComponent<Billboard>();
+
+                    data.gameObject.name = DaggerfallWorkshop.Utility.GameObjectHelper.GetGoFlatName(archive, record);
+                    
+                    //Dungeon mode needs us to update the rdbObj and building mode ignores it.
+                    data.rdbObj.Resources.FlatResource.TextureArchive = archive;
+                    data.rdbObj.Resources.FlatResource.TextureRecord = record;
+                    
+                    DestroyImmediate(Selection.activeGameObject.GetComponent<Collider>());
+                    billboard.SetMaterial(archive, record);
```

Comment:
Not sure about this change to the flat switching for buildings. That is code from UVs original building editor that I basically left alone. This change seems to mean that the id in the attached WorldDataEditorObject won't be getting updated with the new archive and texture which are subsequently used to save to the JSON.

This is only by reviewing code visually, I have not had a chance to try it out yet. Could you comment?

Reply:
I tested to make sure that the output from saving was not affected by this change for both RDBs and RMBs.
The original "building mode" code saves the new archive ID data with the line "data.id = tempArray[pos];" just above here.
Lines 799 and 800 are from the "dungeon mode" code that you added should update the necessary data saved to RDBs

Lines 673 - 682 in the original code only updates the image displayed on the billboard by creating a new temporary billboard and copying its renderer.  

However, doing it this way didn't update the billboard's Summary.Size.  This is the part that caused objects to reposition when saving.  If an archive started with a large image, and you switched it to a smaller one, it would retain the size data of the first.  When you load, the billboards would be recreated from the json and load with the correct size data which would reposition them.

I decided to replace that with a call to SetMaterial since that will correctly update the image and size data.  Though, since it's intended to be called once on a billboard, it duplicates the collider.  I get around that by just removing the collider before SetMaterial since the editor doesn't really care about it.

Reply:
Yes I tested also and it works. I think there is some redundant code after I made the dungeon editor which had to work differently from UncannyValley's building editor (that was the genesis of the WDE) - I should have tidied up better, but it was a lot more work than I had expected.